Project_One/app/src/main/java/com/justinwells/project_one/Main2Activity.java

```diff
@@ -0,0 +1,136 @@
+                builder.setTitle("Enter Task");
+                final EditText task = new EditText(Main2Activity.this);
+
+                builder.setView(task);
+
+                builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+
+                        if (task.getText().toString().trim().equalsIgnoreCase("")){
+                            Toast.makeText(Main2Activity.this, "Text field cannot be blank",
+                                    Toast.LENGTH_SHORT).show();
+                        }
+                        else {
+                            toDoList.add(new ToDoItem(task.getText().toString()));
+                            recyclerView.getAdapter().notifyItemInserted(toDoList.size() - 1);
+                        }
+
+                    }
+                });
+
+                builder.setNegativeButton("CANCEL", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+
+                    }
+                });
+
+                AlertDialog addTask = builder.create();
+                addTask.show();
+
+            }
+        });
+
+        removeButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(Main2Activity.this);
+                builder.setTitle("Remove Selected Items?");
+
+                builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        for (int i=0; i< toDoList.size(); i++) {
+                            if (toDoList.get(i).isDone()) {
+                                toDoList.remove(i);
+                                //listManager.getList(workingIndex).removeItem(i);
+                                recyclerView.getAdapter().notifyItemRemoved(i);
+                                i--;
+                            }
+                        }
+                        //toDoList.remove(toDoList.size()-1);
+
+                    }
+                });
+
+                builder.setNegativeButton("CANCEL", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+
+                    }
+                });
+
+                AlertDialog removeItem = builder.create();
+                removeItem.show();
+            }
+        });
+        backButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent();
+                setResult(RESULT_OK,intent);
```

Comment:
Since you're not putting any "extras" on this intent, you're not actually passing any data back to the previous activity, so there's really no need to use `startActivityForResult()` and `setResult()` in this situation.